### PR TITLE
Hearing Worksheet Save before PDF

### LINKED ISSUE #3968: [Frontend] Hearing Worksheet | Save before generating PDF
If a user has unsaved edits on the hearing worksheet, they won't be displayed on the PDF. When a user clicks 'Save to PDF', it should call the save function, then generate the PDF. 

AC:
- All edits are displayed on the PDF.

### PR DESCRIPTION
**Pull request by nolawi**
Resolves #3968 
## PR
- [x] Triggers Autosave via click of Save PDF button and open PDF in new Tab

### CODE REVIEW

File: client/app/hearings/HearingWorksheet.jsx [hunk 1]
```diff
@@ -65,6 +65,11 @@ export class HearingWorksheet extends React.PureComponent {
     this.props.toggleWorksheetSaving();
   };
 
+  savePDF = (worksheet, worksheetIssues) => {
```
Comment (sharonwarner) [inline client/app/hearings/HearingWorksheet.jsx]: Can we name this something else so that it's obvious it also opens a new window?

Comment (nolawi) [reply client/app/hearings/HearingWorksheet.jsx]: Updated. LMK if it should be something like `openWorksheetPdf`

**Comment by nolawi** [comment]
I attempted to `use` some sort of promise, but it would have to go to the props, since save is async. Current solution forces a remount on the component before the print window opens.

File: client/app/hearings/HearingWorksheet.jsx [hunk 2]
```diff
@@ -60,9 +60,20 @@ export class HearingWorksheet extends React.PureComponent {
...
-    this.props.saveWorksheet(worksheet);
     this.props.saveIssues(worksheetIssues);
+    this.props.saveWorksheet(worksheet);
     this.props.toggleWorksheetSaving();
+  }
+
+  openPdf = (worksheet, worksheetIssues) => {
+    let print = window.open(`${window.location.pathname}/print`, '_blank');
+
+    if (this.props.worksheet.edited) {
```
Comment (sharonwarner) [inline client/app/hearings/HearingWorksheet.jsx]: This won't work if somebody has only edited an issue. We set the edited flag on the issue directly.

File: client/app/hearings/HearingWorksheet.jsx [hunk 3]
```diff
@@ -60,9 +60,20 @@ export class HearingWorksheet extends React.PureComponent {
...
+      this.save(worksheet, worksheetIssues)();
+      window.location.reload();
```
Comment (sharonwarner) [inline client/app/hearings/HearingWorksheet.jsx]: Does this definitely work? does 'window.location.reload()' wait for 'this.save(worksheet, worksheetIssues)()' to finish?

File: client/app/hearings/HearingWorksheet.jsx (hunk 2 above)
Comment (nolawi) [reply client/app/hearings/HearingWorksheet.jsx]: Yes, I can add a check for the `edited flag` on worksheetissues; OR all edits including issues should have a flag on the worksheet flag. I am working on it now

File: client/app/hearings/HearingWorksheet.jsx (hunk 3 above)
Comment (nolawi) [reply client/app/hearings/HearingWorksheet.jsx]: `this.save` is automatic. The wait is on the rerender of the component. The print page needs rendered component.

**Comment by nolawi** [comment]
Closing temporarily

**Pull request closed without merging**